pubpub/pubpub
# citations real time

## Issue(s) Resolved
#2408 

## Test Plan

### To Authenticate

0. log in
1. create you citations-testing pub
2. create a citation
3. in the citation editor, if you haven't yet authenticated, you'll see this button: 
![Screenshot from 2023-03-23 16-11-45](https://user-images.githubusercontent.com/14115194/227341016-b633cbe6-c0d3-47f2-bf98-e9f9f0c5d5f3.png)
4. click 'Connect Zotero', which will take you to your privacy settings page
5. click 'Activate' in the zotero integration section
6. complete the zotero-side oauth flow using login credentials available from 1Password until you are redirected to a 404 pubpub.org url (something along the lines of `https://www.pubpub.org/auth/zotero/redirect?oauth_token=c789475127776014731f&oauth_verifier=fae9f767121206365c3b`)
7. edit your url, replacing the part up 'til `pubpub.org` with the review app url (`https://pubpub-pipel-bh-citatio-4arxsu.herokuapp.com/`) and then hit enter
8. you should be returned to the privacy settings tab and the zotero integration section should offer you the choice to remove the integration:
![Screenshot from 2023-03-23 16-17-32](https://user-images.githubusercontent.com/14115194/227342521-f18604a9-26ad-4ec6-a53e-962e425b353b.png)

### Once Authenticated

0. make sure zotero integration is active
10. open your citation pub
11. create/click on your citation to open the editor
13. type a search query into zotero search prompt box
14. after 300ms debounce, request is fired + loading state
15. one of the suggested results, should produce formatted structured citation below
16. click 'update citation' -> voila

## Screenshots (if applicable)
Without integration/log-in (above)
With integration
![Screenshot from 2023-03-23 16-20-13](https://user-images.githubusercontent.com/14115194/227343315-5af2253b-3592-4bd1-af01-d600c1f38a35.png)

After search
![Screenshot from 2023-03-23 16-20-28](https://user-images.githubusercontent.com/14115194/227343336-523e391b-4d61-4eb2-a320-6c2f6fdc187c.png)

### Notes/Context/Gotchas

### Supporting Docs
[spec](https://notes.knowledgefutures.org/pub/ck6f4zsx#alternative-live-search-results)
